### ISSUE TITLE
Doesn't work

### ISSUE THREAD

**Issue by svein83** [harryallerston/RPI-Wireless-Hotspot#22]
This does not work at all on a newly written sd card with Jessie or Stretch.

After Install, it is lost from LAN, and it can't be reconnected.

**Comment by T-minus** [comment]
I have the same problem. The only thing I haven't tried is to install this and restart prior to running apt-get update. Its habit to update right out of the box.

**Comment by clem9669** [comment]
Similar issue ... 👍

**Comment by vanderblugen** [comment]
I'm having the same reaction.  I'm following [Know-How](https://twit.tv/shows/know-how/episodes/301) and having no luck.  I think that I'm going to abandon it for now considering I can't get it to work.

**Comment by T-minus** [comment]
Yea, I tried several times and also abandoned this one. I did several searches and found other scripts that do in fact work with Jesse. @vanderblugen I was also following the same tutorial. I screwed up on tor after I found a different wifi hotspot script and have not tried to redo it since. Once I get a little more time I will re-attempt it.

**Comment by vanderblugen** [comment]
@T-minus, I just opened another issue that technically does fit this, but wanted to bring more attention to it and outline exactly the flow that I was following.

**Comment by unixabg** [comment]
Greetings,
Sorry for long delay on response. I setup a small test today to examine the issue on Stretch, have not tested on Jessie, however it should still be all good on Jessie. Now for your issue: What happens if you run:

sudo ifup eth0

and or add the below line to in /etc/network/interfaces (perhaps above auto wlan0 entry)

auto eth0 

then reboot.

**Comment by vanderblugen** [comment]
I actually just did a fresh install.  I don't see that anyone updated the files, I may have missed it though.  I am able to pull internet on the Raspberry after installing fresh and reinstalling the hotspot, but this time I'm not getting an IP address for the connecting device.

**Comment by vanderblugen** [comment]
Quick update...  I know the next question is going to be to check the dhcp service.
I did check the service `service udhcpd status` and it showed "active exited".
So if I restart the service `sudo service udhcpd restart` and I pull up the status again it shows "active (running)".  I can add a delay to the service starting by modifying the /etc/udhcpd.conf file and adding `sleep 10` after the comments but after a reboot the service is "active exited" and won't assign the IP.
Suggestions?

**Comment by unixabg** [comment]
Greetings,
I think is on the right track and I am confident it is a race condition with udhcpd service starting before the wlan0 interface is ready. Try running:
         `journalctl -xl | grep udhcpd`
and look for something like:
         `raspberrypi udhcpd[378]: is interface wlan0 up and configured?: Cannot assign requested address`
I am working on a least intrusive and clean way to address the delaying the startup of udhcpd. Ideas are also welcome.

**Comment by vanderblugen** [comment]
I actually ran a full reimage and reinstall again.  This time everything worked.  I made sure to note my exact flow of events.  
Fresh install of 2017-09-07-raspbian-stretch.img,  Paired Bluetooth Mouse, Changed default password, Updated hostname, Updated keyboard layout, Expanded file system, Rebooted.  
Then cloned the RPI Hotspot and installed it.  Then rebooted.  After that it seems to work without issue.
Could possibly be a reboot in the middle.  But I am not sure.

**Comment by shortphaselabs** [comment]
What's the process to get this working again? Still having trouble here with pi zero w and USB WiFi dongle.

**Comment by uilfut** [comment]
Yeah once this install is run it doesn't appear on the router (ethernet) anymore, and you can't connect to it's wireless network it created. Wish I'd read here before trying!

Fresh stretch install on Pi3B.

**Comment by unixabg** [comment]
Greetings,
Could you clone and try this repository
https://github.com/unixabg/RPI-Wireless-Hotspot
and see if all works for you? ty

**Comment by uilfut** [comment]
Thanks, no luck. I'll try setting it up manually. Thanks for reaching out.

I'm not sure if it's a wireless driver issue. I'm using the built-in wifi on the 3B. Some of the documentation has mentioned different drivers for each card, specified in hostapd.conf.

**Comment by unixabg** [comment]
Greetings,
The link to the repo I pointed to has been tested and should work with the 3B. I have a 3B+ coming and will be testing against that soon as well. Did you do a pristine latest raspbian install + the my repo?

**Comment by unixabg** [comment]
Greetings,
Ok just tested 3B and 3B+ with pristine stretch 2018-03-13-raspbian-stretch-lite.img and https://github.com/unixabg/RPI-Wireless-Hotspot. The 3B all works as it should. However on 3B+ all installed but would not launch hostapd. The solution appears to be to run:
`sudo rfkill unblock wifi`
Then reboot. After reboot all seems well on my test environment. The above information was found here https://www.raspberrypi.org/forums/viewtopic.php?t=208597

**Comment by DracoDruid** [comment]
I just tried to get this going on a freshly installed Raspbian stretch lite (2018-06-27) and my new Pi 3 B+.
After the reboot, I had to failure messages concerning "dhcpcd" and "Raise Network Interfaces".
Your last hint didn't work for me either, sadly.
I am now reflashing Raspbian and try again.
Have you had any new insight into this problem?
